packages/fx-core/src/core/middleware/projectMigratorV3.ts

```diff
@@ -179,11 +179,11 @@ export const ProjectMigratorMWV3: Middleware = async (ctx: CoreHookContext, next
     );
     ctx.result = err(AbandonedProjectError());
     return;
-  } else if (versionForMigration.state === VersionState.upgradeable && checkMethod(ctx)) {
-    if (!checkUserTasks(ctx)) {
-      ctx.result = ok(undefined);
-      return;
-    }
+  }
+  const projectPath = getParameterFromCxt(ctx, "projectPath", "");
+  const isValid = await checkActiveResourcePlugins(projectPath);
+  const isUpgradeable = isValid && versionForMigration.state === VersionState.upgradeable;
+  if (isUpgradeable) {
```

Comment:
misses `checkMethod(ctx)` in original if condition

Reply:
CheckMethod condition is correct in v5. 
It excludes  getProjectConfig and checkPermission which are needed to add upgrade logic. So remove the condition.